### PR TITLE
Update api-version for DPS preview

### PR DESCRIPTION
**Pull request by pierreca**
Update api-version from `2018-09-01-preview` to `2018-11-01`

### CODE REVIEW

**Comment by pierreca** [comment]
@anthonyvercolano please take a second look. I had to update the attestation API.

File: provisioning/service/test/_provisioningserviceclient_test.js [hunk 1]
```diff
@@ -648,7 +648,30 @@ describe('ProvisioningServiceClient', function () {
       };
 
       var de = new ProvisioningServiceClient({ host: 'host', sharedAccessSignature: 'sas' }, fakeHttpHelper);
-      de.getAttestationMechanism(testEnrollmentId, testCallback);
+      de.getIndividualEnrollmentAttestationMechanism(testEnrollmentId, testCallback);
+    });
+  });
+
+  describe('#getEnrollmentGroupAttestationMechanism', function () {
```
Comment (anthonyvercolano) [inline provisioning/service/test/_provisioningserviceclient_test.js]: >describe [](start = 2, length = 8)

NIT, big nit: For both the individual and for the group I would have included the requirement item.

File: provisioning/service/src/provisioningserviceclient.ts [hunk 2]
```diff
@@ -297,6 +299,34 @@ export class ProvisioningServiceClient {
...
+  public getEnrollmentGroupAttestationMechanism(enrollmentGroupId: string): Promise<ResultWithHttpResponse<AttestationMechanism>>;
+  public getEnrollmentGroupAttestationMechanism(enrollmentGroupId: string, callback?: HttpResponseCallback<AttestationMechanism>): Promise<ResultWithHttpResponse<AttestationMechanism>> | void  {
+    /*Codes_SRS_NODE_PROVISIONING_SERVICE_CLIENT_16_003: [The `getEnrollmentGroupAttestationMechanism` method shall throw a `ReferenceError` if the `enrollementGroupId` parameter is falsy.]*/
+    if (!enrollmentGroupId) {
+      throw new ReferenceError('enrollmentGroupId cannot be \'' + enrollmentGroupId + '\'');
+    }
+
+    /*Codes_SRS_NODE_PROVISIONING_SERVICE_CLIENT_16_004: [The `getEnrollmentGroupAttestationMechanism` shall construct an HTTP request using information supplied by the caller as follows:
+    ```
+    POST /enrollmentgroups/<encodeUriComponentStrict(enrollmentGroupId)>/?api-version=<version> HTTP/1.1
```
Comment (anthonyvercolano) [inline provisioning/service/src/provisioningserviceclient.ts]: >(enrollmentGroupId)>/?api-version=<version> [](start = 52, length = 43)

Nit: for this group and for the individual the requirements do NOT include the /attestationmechanism text.  Like I said, just a NIT.

**Comment by anthonyvercolano** [comment]
Consider this codeflow APPROVED a second time.

---
In reply to: [444742155](https://github.com/Azure/azure-iot-sdk-node/pull/441#issuecomment-444742155) [](ancestors = 444742155)

**Comment by pierreca** [comment]
@anthonyvercolano thanks! build might fail again since the fix for the service is still being deployed across all regions. by next week we'll be able to re-run it, validate and merge. no action for you at this point :)

**Pull request merged**